test/jdk/java/awt/EventQueue/PushPopDeadlock/PushPopDeadlock.java

```diff
@@ -0,0 +1,94 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4212687
+ * @summary Verifies that calling EventQueue.push() and EventQueue.pop()
+ *          does not deadlock.
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual PushPopDeadlock
+ */
+
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Label;
+import java.awt.Robot;
+import java.awt.Toolkit;
+
+public class PushPopDeadlock {
+    static int counter = 0;
+    static Robot robot;
+
+    public static void main(String[] args) throws Exception {
+        robot = new Robot();
+        String INSTRUCTIONS = """
+                Click rapidly in the Frame labeled 'Click Here!'.
+                The number in the Frame should continue to increase. If the number
+                stops increasing (remains at a constant value), the test fails.
+                """;
+
+        PassFailJFrame.builder()
+                .title("Test Instructions")
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 2)
+                .columns(35)
+                .testUI(PushPopDeadlock::createUI)
+                .build()
+                .awaitAndCheck();
+    }
+
+    public static Frame createUI() {
+        Frame f = new Frame("Click Here!");
+
+        Label l = new Label("" + counter);
+        f.add(l);
+
+        EventQueue q = new EventQueue() {
+            public void push(EventQueue queue) {
+                super.push(queue);
+                pop();
+            }
+        };
+        EventQueue q2 = new EventQueue();
+
+        f.setSize(200, 200);
+
+        Toolkit.getDefaultToolkit().getSystemEventQueue().push(q);
+
+        new Thread(() -> {
+            while (true) {
+                robot.delay(500);
+                l.setText("" + ++counter);
+                q.push(q2);
+                try {
+                    Thread.currentThread().sleep(500);
+                } catch (InterruptedException e) {
+                    return;
+                }
+            }
```

Comment:
This is a busy wait loop, maybe there is a better way of doing it although it might also be out-of-scope for this PR changes.

Reply:
I'll leave it for a future PR if needed.